src/api/code-challenge/content-types/code-challenge/lifecycles.js

```diff
@@ -1,43 +1,131 @@
 
-// runMetaTestsNeed more specific name later, maybe use "validateInternalTests"
-async function runMetaTests(eventTests, eventMetaTests) {
+const conductMetaTestsOnTests = async (
+  eventTests,
+  eventMetaTests,
+  challengeLabel
+) => {
   const [internalTests, metaTests] = await Promise.all([
     getInternalTests(eventTests),
     getMetaTests(eventMetaTests),
   ]);
+  for (let i = 0; i < metaTests.length; i++) {
+    let metaTestResult = await executeTests(
+      metaTests[i].caseCode,
+      internalTests,
+      metaTests[i].label,
+      challengeLabel,
+      metaTests[i].id,
+      metaTests[i].passes
+    );
 
-  // console.log("final internalTests: ", internalTests);
-  // console.log("final metaTests: ", metaTests);
+    // MetaTest Final Results being logged to Console.
+    await console.log(
+      `\n\n<"${metaTests[i].label}" METATEST #${metaTests[i].id}> \n\n`,
+      metaTestResult,
+      "\n\n"
+    );
+  }
+};
 
-  // For next part of feature: return some type of eval using internalTests & metaTests
-  return false;
-}
+const executeTests = async (
+  metaCaseCode,
+  internalTests,
+  metaLabel,
+  challengeLabel,
+  metaTestId,
+  evalResultShouldBe
+) => {
+  return pMap(
+    internalTests?.filter(removeEmpty) || [],
+    async (internalTest) => {
+      const {
+        label: internalTestLabel,
+        internalTestCode: internalTestCode,
+        id: internalTestId,
+      } = internalTest;
+
+      // console.log("<internalTest pMap> ", internalTest);
+      // console.log("newTest name: ", newTest);
+
+      const testEvaluatorResults = await runTestEvaluator({
+        metaCaseCode,
+        internalTestCode,
+        metaLabel,
+        challengeLabel,
+        metaTestId,
+        internalTest,
+        evalResultShouldBe,
+      });
+      return {
+        short: {
+          description: null,
+          ...pick(testEvaluatorResults, ["description"]),
+        },
+        long: {
+          challengeLabel,
+          metaTestId,
+          internalTestId,
+          internalTestLabel,
+          userPassed: null,
+          evalResultType: null,
+          ...pick(testEvaluatorResults, [
+            "evalResultType",
+            "userPassed",
+            "evalError",
+          ]),
+        },
+      };
+    }
+  );
+};
 
 async function getInternalTests(eventTests) {
   const internalTests = await strapi.db
     .query("challenge.code-challenge-test")
     .findMany({
-      select: ["id", "internalTest"],
+      select: ["id", "internalTest", "label"],
       where: {
         id: eventTests.map(({ id: testId }) => testId),
       },
     });
-  return internalTests;
+
+  // Originally, the content of internalTest === internalTestCode in strapi.
+  // I renamed internalTest to internalTestCode for clarity in lifecycles.js & test-evaluator.ts
+  // This renamed variable makes: (internalTestCode vs internalTestLabel vs internalTestId) the children of internalTestPackage
```

Comment:
nit. use multi line comments instead of single line

Reply:
This was due to auto-format. What auto-format settings should we use for this project?